cmds/core/md5sum/md5sum.go

```diff
@@ -7,84 +7,91 @@ package main
 
 import (
 	"crypto/md5"
+	"flag"
 	"fmt"
 	"io"
 	"log"
 	"os"
 
-	"github.com/spf13/pflag"
+	"github.com/u-root/u-root/pkg/uroot/util"
 )
 
-func getInput() (input []byte, err error) {
-	return io.ReadAll(os.Stdin)
-}
+var (
+	help    = flag.Bool("h", false, "Show this help and exit")
+	version = flag.Bool("v", false, "Print Version")
+)
+
+var usage = "md5sum: md5sum <File Name>"
 
-func helpPrinter() {
-	fmt.Printf("Usage:\nmd5sum <File Name>\n")
-	pflag.PrintDefaults()
-	os.Exit(0)
+func getInput(r io.Reader) (input []byte, err error) {
+	return io.ReadAll(r)
 }
 
 func versionPrinter() {
 	fmt.Println("md5sum utility, URoot Version.")
-	os.Exit(0)
 }
 
-func calculateMd5Sum(fileName string, data []byte) string {
+func calculateMd5Sum(w io.Writer, fileName string, data []byte) (string, error) {
 	if len(data) > 0 {
-		return fmt.Sprintf("%x", md5.Sum(data))
+		return fmt.Sprintf("%x", md5.Sum(data)), nil
 	}
 
 	fileDesc, err := os.Open(fileName)
 	if err != nil {
-		log.Fatal(err)
+		return "", err
 	}
 	defer fileDesc.Close()
 
 	md5Generator := md5.New()
 	if _, err := io.Copy(md5Generator, fileDesc); err != nil {
-		log.Fatal(err)
+		return "", err
 	}
 
-	md5Sum := fmt.Sprintf("%x", md5Generator.Sum(nil))
-	return md5Sum
+	_, err = fmt.Fprintf(w, "%x", md5Generator.Sum(nil))
+	if err != nil {
+		return "", err
+	}
+	return "", nil
 }
-
-func main() {
+func md5Sum(w io.Writer, r io.Reader, args ...string) error {
 	var (
-		help    bool
-		version bool
-		input   []byte
-		err     error
+		input []byte
+		err   error
 	)
+
 	cliArgs := ""
-	pflag.BoolVarP(&help, "help", "h", false, "Show this help and exit")
-	pflag.BoolVarP(&version, "version", "v", false, "Print Version")
-	pflag.Parse()
 
-	if help {
-		helpPrinter()
+	if *help {
+		util.Usage(usage)
+		flag.Usage()
+		return nil
 	}
 
-	if version {
+	if *version {
```

Comment:
This versionprinter stuff is something our other commands do, I'm wondering if we have to have it at all.

Overall, this program feels like an awkward fit to the rest of u-root style. 

Would it be possible to just take, e.g., our cat command, and make the small change it would need to it to replace this code entirely? This program has a style that grates on my eyeballs a little.

Reply:
I removed the version printer function for now.